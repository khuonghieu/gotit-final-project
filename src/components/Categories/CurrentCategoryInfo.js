import React, { useState } from 'react';
import { connect } from 'react-redux';
import { viewCategory } from '../../actions/categories';

function mapStateToProps(state) {
  return {
    categoryId: state.categories.currentCategory,
  };
}

const mapDispatchToProps = (dispatch) => ({
  viewCategory: (categoryId) => dispatch(viewCategory(categoryId)),
});

export function CurrentCategoryInfo({ categoryId, viewCategory }) {
  const [categoryInfo, setCategoryInfo] = useState({});

  // Must use React.useEffect, else the unit test wont work
  React.useEffect(() => {
    (async () => {
      const { payload } = await viewCategory(categoryId);
      setCategoryInfo(payload);
    })();
  }, [categoryId, viewCategory]);

  return (
    <div>
      {categoryInfo ? (
        <p>
          category name:
          {' '}
          {categoryInfo.name}
          <br />
          category desc:
          {' '}
          {categoryInfo.description}
        </p>
      ) : 'No category to show'}
    </div>
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(CurrentCategoryInfo);
