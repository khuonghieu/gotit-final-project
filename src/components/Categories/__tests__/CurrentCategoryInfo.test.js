import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CurrentCategoryInfo } from '../CurrentCategoryInfo';

configure({ adapter: new Adapter() });

describe('components/Categories/CurrentCategoryInfo.js', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    jest.spyOn(React, 'useEffect').mockImplementation((f) => f());
    props = {
      categoryId: null,
      viewCategory: jest.fn(),
    };
  });
  const setup = () => {
    wrapper = shallow(
      <CurrentCategoryInfo {...props} />,
    );
  };
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should get category info when change categoryId', () => {
    setup();
    wrapper.setProps({ ...props, categoryId: 1 });
    expect(props.viewCategory).toHaveBeenCalled();
  });
});
