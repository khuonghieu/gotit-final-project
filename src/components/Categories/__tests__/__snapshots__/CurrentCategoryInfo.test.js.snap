// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Categories/CurrentCategoryInfo.js should render correctly 1`] = `ShallowWrapper {}`;
